jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -65,4 +84,41 @@ private static String determineJarType(Path jarPath) throws IOException {
       return STANDARD;
     }
   }
+
+  /**
+   * Determines the java version of JAR. Derives the version from the first .class file it finds in
+   * the JAR.
+   *
+   * @param jarPath path to the jar
+   * @return java version
+   * @throws IOException if I/O exception thrown when opening the jar file
+   */
+  @VisibleForTesting
+  static Integer getJavaMajorVersion(Path jarPath) throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      List<String> jarEntries =
+          jarFile.stream().map(JarEntry::toString).collect(Collectors.toList());
```

Comment:
I realized that using .nextElement() results in a `NoSuchElementException` if there are no more elements - this would cause an issue if an empty jar was passed in.

Reply:
But you were checking `hasMoreElements()`. It will never fail.

I liked the previous code with `Enumeration`. Although performance-wise, I think there's zero impact, but it allowed short-circuiting. (Now `stream().collect()` needs to process all elements.)

Reply:
Oh no, my bad. For some reason, I had an extra `.nextElement()` in the while loop when I was testing this locally. Sorry about the confusion! Reverting this change.